<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>532</width>
    <height>363</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Password Generator</string>
  </property>
  <property name="windowIcon">
   <iconset resource="resources.qrc">
    <normaloff>:/icons/icons/ic_vpn_key_128_28771.ico</normaloff>:/icons/icons/ic_vpn_key_128_28771.ico</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidge">
   <property name="styleSheet">
    <string notr="true">QWidget{
    background-color: #673AB7;
    color: white;
    font-family: Verdana;
    font-size: 16pt;
    margin: 10px;
}

QPushButton{
    border: 2px solid #2196F3;
    border-radius: 5px;
}

QPushButton#btn_lower, #btn_upper, #btn_digits, #btn_symbols{
    padding: 10px;
}

QPushButton:hover {
    border-color: #2196F3;
}

QPushButton:pressed{
    border: 4px solid #2196F3;
    border-radius: 5px;
}

QPushButton:checked{
    background-color: #1A237E;
    border-color: #2196F3;
}</string>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="3" column="0">
     <layout class="QHBoxLayout" name="layout_length">
      <item>
       <widget class="QSlider" name="slider_length">
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="styleSheet">
         <string notr="true">QSlider::groove:horizontal{
    background-color: transparent;
    height: 5px;
}
QSlider::sub-page:horizontal{
    background-color: #2196F3;
}

QSlider::add-page:horizontal{
    background-color: gray;
}

QSlider::handle:horizontal{
    background-color: white;
    width: 22px;
    border-radius: 10px;
    margin-top: -8px;
    margin-bottom: -8px;
}
</string>
        </property>
        <property name="maximum">
         <number>80</number>
        </property>
        <property name="value">
         <number>15</number>
        </property>
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinbox_length">
        <property name="styleSheet">
         <string notr="true">QSpinBox{
    border:2px solid #2196F3;
    border-radius: 5px;
    background: transparent;
    padding: 2px;
}

QSpinBox:hover{
    border-color: #2196F3;
}</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::ButtonSymbols::NoButtons</enum>
        </property>
        <property name="maximum">
         <number>80</number>
        </property>
        <property name="value">
         <number>15</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="4" column="0">
     <layout class="QHBoxLayout" name="layout_character">
      <item>
       <widget class="QPushButton" name="btn_lower">
        <property name="text">
         <string>a-z</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_upper">
        <property name="text">
         <string>A-Z</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_digits">
        <property name="text">
         <string>0-9</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_symbols">
        <property name="text">
         <string>#$%</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <widget class="QFrame" name="layot_password">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <layout class="QHBoxLayout" name="frame_password">
       <item>
        <widget class="QFrame" name="frame_password_2">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="styleSheet">
          <string notr="true">QFrame{
    border: 2px solid gray;
    border-radius: 5px;
    margin-left: 0;
}

QFrame{
    border-color: #2196F3;
}</string>
         </property>
         <property name="frameShape">
          <enum>QFrame::Shape::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Shadow::Raised</enum>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLineEdit" name="lineEdit"/>
          </item>
          <item>
           <widget class="QPushButton" name="btn_visibility">
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="styleSheet">
             <string notr="true">QPushButton{
    border: none;
    margin: 0 ;
    background-color: transparent;
}</string>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="icon">
             <iconset resource="resources.qrc">
              <normaloff>:/icons/icons/eye_slash_visible_hide_hidden_show_icon_145987.ico</normaloff>
              <normalon>:/icons/icons/eye_visible_hide_hidden_show_icon_145988.ico</normalon>:/icons/icons/eye_slash_visible_hide_hidden_show_icon_145987.ico</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>30</width>
              <height>30</height>
             </size>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btn_refresh">
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="styleSheet">
          <string notr="true">QPushButton{
    margin-right: 0;
    margin-left: 0;
    border-color: #2196F3;
}</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="resources.qrc">
           <normaloff>:/icons/icons/twocirclingarrows1_120592.ico</normaloff>:/icons/icons/twocirclingarrows1_120592.ico</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>52</width>
           <height>52</height>
          </size>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btn_copy">
         <property name="cursor">
          <cursorShape>PointingHandCursor</cursorShape>
         </property>
         <property name="styleSheet">
          <string notr="true">QPushButton{
    padding: 5px;
    margin-left: 0;
	border-color:  #2196F3;
}</string>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="icon">
          <iconset resource="resources.qrc">
           <normaloff>:/icons/icons/clipboard_89387.ico</normaloff>:/icons/icons/clipboard_89387.ico</iconset>
         </property>
         <property name="iconSize">
          <size>
           <width>42</width>
           <height>42</height>
          </size>
         </property>
         <property name="shortcut">
          <string>Ctrl+C</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0">
     <layout class="QHBoxLayout" name="layout_info">
      <item>
       <widget class="QLabel" name="label_strength">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignmentFlag::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item row="1" column="0">
     <layout class="QVBoxLayout" name="layout_btn_generate">
      <item>
       <widget class="QPushButton" name="btn_generate">
        <property name="styleSheet">
         <string notr="true">QPushButton{
    background-color: #1A237E;
}</string>
        </property>
        <property name="text">
         <string>Сгенерировать</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources>
  <include location="resources.qrc"/>
 </resources>
 <connections/>
</ui>
